napari/layers/utils/layer_utils.py

```diff
@@ -684,58 +684,201 @@ def features_to_pandas_dataframe(features: Any) -> pd.DataFrame:
+        return _features_to_properties(self._defaults)
+
+    def set_currents(
+        self,
+        currents: Dict[str, np.ndarray],
+        *,
+        update_indices: Optional[List[int]] = None,
+    ):
+        """Sets the default values using the deprecated current properties dictionary.
+
+        May also update some of the feature values to be equal to the new default values.
+        """
+        currents = coerce_current_properties(currents)
+        self._defaults = _validate_features(currents, num_data=1)
+        if update_indices is not None:
+            for k in self._defaults:
+                self._values[k][update_indices] = self._defaults[k][0]
+
+    def resize(
+        self,
+        size: int,
+    ):
+        """Resize this padding with default values if required.
+
+        Parameters
+        ----------
+        size : int
+            The new size (number of rows) of the features table.
+        """
+        current_size = self._values.shape[0]
+        if size < current_size:
+            self.remove(range(size, current_size))
+        elif size > current_size:
+            to_append = pd.concat([self._defaults] * (size - current_size))
+            self.append(to_append)
+
+    def append(self, to_append: pd.DataFrame):
+        """Append new feature rows to this.
+
+        Parameters
+        ----------
+        to_append : pd.DataFrame
+            The features to append.
+        """
+        self._values = self._values.append(to_append, ignore_index=True)
+
+    def remove(self, indices: Any):
+        """Remove rows from this by index.
+
+        Parameters
+        ----------
+        indices : Any
+            The indices of the rows to remove. Must be usable as the labels parameter
+            to pandas.DataFrame.drop.
+
+        Returns
+        -------
+        pd.DataFrame
+            The resulting features table, which contain copies of the existing data.
+        """
+        self._values = self._values.drop(labels=indices, axis=0).reset_index(
+            drop=True
+        )
+
+    def reorder(self, order: Sequence[int]):
+        """Reorders the rows of the feature values table."""
+        self._values = self._values.iloc[order].reset_index(drop=True)
+
+    @classmethod
+    def from_layer(
+        cls,
+        *,
+        features: Optional[Union[Dict[str, np.ndarray], pd.DataFrame]] = None,
+        properties: Optional[
+            Union[Dict[str, np.ndarray], pd.DataFrame]
+        ] = None,
+        property_choices: Optional[Dict[str, np.ndarray]] = None,
+        num_data: Optional[int] = None,
+    ) -> _FeatureManager:
+        """Coerces a layer's keyword arguments to a feature manager.
+
+        Parameters
+        ----------
+        features : Optional[Union[Dict[str, np.ndarray], pd.DataFrame]]
+            The features input to a layer.
+        properties : Optional[Union[Dict[str, np.ndarray], pd.DataFrame]]
+            The properties input to a layer.
```

Comment:
Is that the deprecated properties stuff ? 
Should there be a check/assert that not both properties and features are passed in ? 
Maybe a note on deprecation ?

Reply:
See comment: https://github.com/napari/napari/pull/3904/files#r782328666